gdb/printing.py

```diff
@@ -261,20 +261,23 @@ def _get_children(self, max_count: int) -> None:
                     (c.GetName(), gdb.Value(c))
                     for c in self._sbvalue.GetNonSyntheticValue()]
             # Prepend errors (as synthetic children) to the list of children.
+            byte_order = self._sbvalue.GetTarget().GetByteOrder()
+            address_byte_size = self._sbvalue.GetTarget().GetAddressByteSize()
             for error_str in self._captured_errors:
+                data = lldb.SBData.CreateDataFromCString(
+                    byte_order, address_byte_size, error_str)
+                data_type = self._sbvalue.GetTarget().GetBasicType(
+                    lldb.eBasicTypeChar).GetArrayType(len(error_str))
```

Comment:
I just realized that if this is taking the string as a C string, you may need to add 1 to the length returned by Python's `len`. Could you double check before we merge this?

Reply:
The `error_str` isn't null terminated. I just added a null terminator at the end of `error_str` to ensure its ending with `\0`.

Reply:
I think SWIG takes care of inserting the null terminator when passing a Python string to a function that takes a C string. The thing I was worried about is size of `data_type`, but looking at the implementation of `CreateDataFromCString` it looks like it makes a copy of the original bytes and keeps track of the length.

So I think removing the `error_str += '\0'` line is fine. Once you do that this should be ready to merge. Sorry for the confusion.